src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -47,35 +64,58 @@ Status ledgerFromRequest (
 
     if (!hashValue.empty())
     {
+        if (! hashValue.isString ())
+            return {rpcINVALID_PARAMS, "ledgerHashNotString"};
+
         uint256 ledgerHash;
-        if (hashValue.isString() && ledgerHash.SetHex (hashValue.asString ()))
-            ledger = netOps.getLedgerByHash (ledgerHash);
-        else
+        if (! ledgerHash.SetHex (hashValue.asString ()))
             return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
+           
+        if ((ledger = netOps.getLedgerByHash (ledgerHash)) == nullptr)
+        {
+            auto cl = netOps.getCurrentLedger ();
+            if (cl == nullptr || ! synced (netOps, cl))
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
+        }
     }
     else if (indexValue.isNumeric())
     {
-        ledger = netOps.getLedgerBySeq (indexValue.asInt());
+        if ((ledger = netOps.getLedgerBySeq (indexValue.asInt ())) == nullptr)
+        {
+            auto cl = netOps.getCurrentLedger ();
+            if (cl == nullptr || ! synced (netOps, cl))
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
+        }
     }
     else
     {
-        auto index = indexValue.asString();
-        auto isCurrent = index.empty() || index == "current";
+        auto const index = indexValue.asString ();
```

Comment:
Thanks for the feedback. I need eyes on how the logic might adversely affect RPC calls from clients. Obviously the intent is to reduce the needless current and closed ledger requirements which would tremendously improve perceived client handler stability and performance. But lets make sure we don't inadvertently spoil the effect.